packages/fxa-content-server/app/scripts/views/behaviors/navigate-or-redirect.js

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A behavior that navigates to the relier's redirectTo value if present, or navigates to a fixed endpoint
+ */
+
+import _ from 'underscore';
+import AuthErrors from '../../lib/auth-errors';
+import NavigationBehavior from './navigate.js';
+
+const NavigateOrRedirectBehavior = function(endpoint, options = {}) {
+  const behavior = function(view, account) {
+    let redirectTo = view.relier.get('redirectTo');
+    if (redirectTo) {
+      redirectTo = new URL(redirectTo, location.href);
+      if (redirectTo.origin !== location.origin) {
+        const err = AuthErrors.toError('INVALID_REDIRECT_TO');
+        this.logError(err);
+        throw err;
+      }
+      redirectTo = redirectTo.pathname + redirectTo.search;
```

Comment:
Open redirects seem a little scary, which is why I put this extra logic here (and below) to at least make sure the redirect can't leave the server.

I guess they should probably be a single function somewhere, I'll try to fix that tomorrow.

Reply:
Huh... but where? `app/scripts/lib/validate.js` ? I can't decide.

Reply:
`validate.js` looks like as good a place as any, IMO

Reply:
Well... I ended up taking out one of the instances of redirection (instead always using behaviors) so I guess since there's only once instance reuse is a moot point, right? 🤷‍♂